src/entity/spatial/Geometry.cpp

```diff
@@ -24,27 +24,20 @@ Geometry::Geometry(const core::IDGenBase* idgen)
     : Entity(idgen)
 {
     this->setDiscriminator<Geometry>();
-
-    //factory geometry creation by sub classes!
 }
 
 Geometry::~Geometry() 
 {
-    //factory geometry delete by sub classes!
 }
 
-const geom::Geometry* Geometry::geometry() 
+geom::Geometry* Geometry::geometry() 
 { 
-    return geometry_;
+    return NULL; // have to be override by childs!
 }
 
-void Geometry::setGeometry(geom::Geometry* geometry) 
-{ 
-    if (geometry != NULL)
-    {
-        factory_->destroyGeometry(geometry_);
-        geometry_ = geometry;
-    }
+const geom::Geometry* Geometry::getGeometry()
+{
+    return getGeometry();
```

Comment:
To quote my IDE:
> all paths through this function will call itself [-Winfinite-recursion]

:wink:

Reply:
Sorry for this one :-|
Fixed in 39b7cc004bcc9487075d08964d3ba72985e0e2ba